Core/src/Detector/detail/CuboidalDetectorHelper.cpp

```diff
@@ -0,0 +1,389 @@
+    for (auto [im, mergeValue] : enumerate(possibleValues)) {
+      // Skip the bin value itself, fusing will took care of that
+      if (mergeValue == bValue) {
+        continue;
+      }
+      // Record the merge splits
+      mergeSplits[im].push_back(2 * boundValues[bValue]);
+      mergeHalfLengths[im] += boundValues[bValue];
+    }
+  }
+
+  // Loop to create the new portals as portal replacements
+  std::vector<PortalReplacement> pReplacements;
+  for (auto [im, mergeValue] : enumerate(possibleValues)) {
+    // Skip the bin value itself, fusing took care of that
+    if (mergeValue == bValue) {
+      continue;
+    }
+
+    // Create the new RecangleBounds
+    // - there are conventions involved, regarding the bounds orientation
+    // - This is an anticyclic swap
+    bool mergedInX = true;
+    switch (bValue) {
+      case binZ: {
+        mergedInX = (mergeValue == binY);
+      } break;
+      case binY: {
+        mergedInX = (mergeValue == binX);
+      } break;
+      case binX: {
+        mergedInX = (mergeValue == binZ);
+      } break;
+      default:
+        break;
+    }
+
+    // The stitch boundarieS for portal pointing
+    std::vector<ActsScalar> stitchBoundaries;
+    stitchBoundaries.push_back(-mergeHalfLengths[im]);
+    for (auto step : mergeSplits[im]) {
+      stitchBoundaries.push_back(stitchBoundaries.back() + step);
+    }
+
+    for (auto [is, index] : enumerate(portalSets[mergeValue])) {
+      // Check if you need to skip due to selections
+      if (!selectedOnly.empty() &&
+          std::find(selectedOnly.begin(), selectedOnly.end(), index) ==
+              selectedOnly.end()) {
+        continue;
+      }
+
+      auto [keepHalfLength, portalTransform] = mergeParameters[index];
+      std::shared_ptr<RectangleBounds> portalBounds =
+          mergedInX ? std::make_shared<RectangleBounds>(mergeHalfLengths[im],
+                                                        keepHalfLength)
+                    : std::make_shared<RectangleBounds>(keepHalfLength,
+                                                        mergeHalfLengths[im]);
+      auto portalSurface =
+          Surface::makeShared<PlaneSurface>(portalTransform, portalBounds);
+      auto portal = std::make_shared<Portal>(portalSurface);
+      // Make the stitch boundaries
+      pReplacements.push_back(
+          PortalReplacement(portal, index, Direction::Backward,
+                            stitchBoundaries, (mergedInX ? binX : binY)));
+    }
+  }
+  // Return proto container
+  DetectorComponent::PortalContainer dShell;
+
+  // Update the portals of all volumes
+  // Exchange the portals of the volumes
+  for (auto& iv : volumes) {
+    ACTS_VERBOSE("- update portals of volume '" << iv->name() << "'.");
+    for (auto& [p, i, dir, boundaries, binning] : pReplacements) {
+      // Fill the map
+      dShell[i] = p;
+      ACTS_VERBOSE("-- update portal with index " << i);
+      iv->updatePortal(p, i);
+    }
```

Comment:
```suggestion
    for (auto& [p, i, dir, boundaries, binning] : pReplacements) {
      (void)dir;
      (void)boundaries;
      (void)binning;

      // Fill the map
      dShell[i] = p;
      ACTS_VERBOSE("-- update portal with index " << i);
      iv->updatePortal(p, i);
    }
```
The compiler doesn't warn for it since it is enough to use 1 variable from an unpacked tuple. But I saw, we indicate at other places and I think it makes the loop a bit clearer.

Reply:
I prefer to keep it, we don't need the extra voiding here, I suppose.